app/components/search/SearchResults/SearchResults.tsx

```diff
@@ -53,30 +56,51 @@ const SearchResults = ({
 
   if (!searchResults) return null;
 
+  const currentPage = searchResults.page ?? 0;
+  const hitsPerPage = searchResults.hitsPerPage ?? 20;
+
   return (
     <div className={styles.searchResultsAndMapContainer}>
       <div
         className={`${styles.searchResultsContainer} ${
           overlayMapWithSearchResults ? styles.overlayMapWithSearchResults : ""
         }`}
       >
-        <div
-          className={`${styles.noResultsMessage} ${
-            hits && hits.length ? styles.hidden : ""
-          }`}
-        >
-          No results found in your area. Try a different location, category, or
-          search term.
-        </div>
-        {hits.map((hit, index) => (
-          <SearchResult
-            hit={hit}
-            index={index}
-            // categoryId={categoryId} // Keep for category ticket
-            key={hit.id}
-          />
-        ))}
-        <ResultsPagination noResults={!hits || !hits.length} />
+        {!hits.length ? (
+          <div
+            className={`${styles.noResultsMessage} ${
+              hits && hits.length ? styles.hidden : ""
+            }`}
+          >
+            <div className={styles.noResultsText}>
+              No results {searchQuery && `for ${` "${searchQuery}" `}`} found in
+              your area.
+              <br /> Try a different location, filter, or search term.
+            </div>
+
+            {searchQuery && <ClearSearchButton />}
+          </div>
+        ) : (
+          <>
+            {searchQuery && (
+              <div className={styles.searchResultsHeader}>
+                <h2>{`${searchResults.nbHits} search results ${
+                  searchQuery && ` for ${searchQuery}`
+                }`}</h2>
+                <ClearSearchButton />
+              </div>
+            )}
+            {hits.map((hit, index) => (
+              <SearchResult
+                hit={hit}
+                index={currentPage * hitsPerPage + index + 1}
```

Comment:
I'm seeing something funky with the search results. If you type in a search query on any other page besides 1, it will reset the url back to page=1 but result 22 will be added to the top of the list with every new search. Occasionally, it will appear above the search results section.

![Screenshot 2024-06-24 at 4 43 06 PM](https://github.com/Exygy/askdarcel-web/assets/61437879/84456031-4252-4556-b303-0dff854c4913)

![Screenshot 2024-06-24 at 5 00 44 PM](https://github.com/Exygy/askdarcel-web/assets/61437879/f1884bcb-e798-459e-b3e9-cf13a2fe2cfc)

Reply:
Good catch! I think the reason is because some of the mock api data has the same id so I changed the key to point to id and name to ensure uniqueness. Couldn't reproduce after that.